lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -134,294 +134,233 @@ public function render_passwordless_method() {
-      </div>
-    <?php
+      $this->render_field_description(
+        __( 'Users migration is disabled. ', 'wp-auth0' ) .
+        __( 'Enabling this exposes migration webservices but the Connection must be updated manually. ', 'wp-auth0' ) .
+        $this->get_docs_link( 'users/migrations/automatic', __( 'More information here', 'wp-auth0' ) )
+      );
     }
-
   }
 
-  public function render_migration_ws_ips_filter() {
-    $v = $this->options->get( 'migration_ips_filter' );
-    $this->render_a0_switch( "wpa0_auth0_migration_ips_filter", "migration_ips_filter", 1, 1 == $v );
+  public function render_migration_ws_ips_filter( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
   }
 
-  public function render_migration_ws_ips() {
-	  $list = $this->options->get( 'migration_ips' );
-	  ?>
-	  <div class="subelement">
-		  <textarea name="<?php echo $this->options->get_options_name(); ?>[migration_ips]" id="wpa0_auth0_migration_ips"><?php echo $list; ?></textarea>
-		  <span class="description">
-	        <?php echo __( 'Only requests from this IPs will be allowed to the migration WS.', 'wp-auth0' ); ?>
-	        <?php echo __( 'Separate multiple IPs with commas.', 'wp-auth0' ); ?>
-        </span>
-	  </div>
-	  <?php
+  public function render_migration_ws_ips( $args ) {
+    $this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Only requests from this IPs will be allowed to the migration WS. ', 'wp-auth0' ) .
+      __( 'Separate multiple IPs with commas', 'wp-auth0' )
+    );
   }
 
-  // TODO: Better documentation for when this should be used
-  public function render_auth0_implicit_workflow() {
-    $v = absint( $this->options->get( 'auth0_implicit_workflow' ) );
-
-    echo $this->render_a0_switch( "wpa0_auth0_implicit_workflow", "auth0_implicit_workflow", 1, 1 == $v );
-?>
-
-    <div class="subelement">
-      <span class="description"><?php echo __( 'Activate this option to change the login workflow and allow the plugin to work when the server doesn\'t have internet access.', 'wp-auth0' ); ?></span>
-    </div>
-    <?php
+  public function render_auth0_implicit_workflow( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Turns on implicit login flow, which most sites will not need. ', 'wp-auth0' ) .
+      __( 'Only enable this if outbound connections to auth0.com are disabled on your server. ', 'wp-auth0' ) .
+      __( 'Your Client should be set to "Single Page App" in your ', 'wp-auth0' ) .
+      $this->get_dashboard_link( 'clients' ) .
+      __( ' for this setting to work properly. ', 'wp-auth0' ) .
+      __( 'This will limit profile changes and other functionality in the plugin', 'wp-auth0' )
```

Comment:
Is there any link you can provide with more details to this "other functionality" that gets limited?

Reply:
Not really. This is a strange setting to begin with but it's in use currently so can't be removed. Basically, instead of pulling the user data from the management API, it's only coming from the `id_token`.